### ISSUE TITLE
XPath result incorrect for first item in repeating group when the there is no matching data in the first node

### ISSUE THREAD

**Issue by pbowen-oc** [OpenClinica/enketo-express-oc#504]
This is present in 2.5.6.oc-12 and the latest master.

Steps:
1. Open the attached form with the attached instance and clinicaldata instance. This form checks the clinicaldata instance event structure to read in the OpenClinica:Current attribute from each of the corresponding nodes in the file. The attached file only has the OpenClinica:Current attribute in the second node.

[cross-form openclinica current.xform.txt](https://github.com/OpenClinica/enketo-express-oc/files/7214745/cross-form.openclinica.current.xform.txt)
[cross-form openclinica current.instance.txt](https://github.com/OpenClinica/enketo-express-oc/files/7214746/cross-form.openclinica.current.instance.txt)
[cross-form openclinica current.clinicaldata-instance.txt](https://github.com/OpenClinica/enketo-express-oc/files/7281295/cross-form.openclinica.current.clinicaldata-instance.txt)

Expect:
The form should report that the first and third occurrences of the event do not have the tag and the second occurrence does have the tag (with value "Yes").

Actual:
The form reports that the the second occurrence has the tag (with value "Yes") and he third occurrence does not have it (as expected), but it reports that the first occurrence does have the tag (with value "Yes").

![Screenshot 2021-09-22 203123](https://user-images.githubusercontent.com/25179620/134446380-4279d0ba-ef1b-43a1-baa1-d251393c363d.png)

Further testing confirmed that the first repeat occurrence is simply reading in the first appearance of the OpenClinica:Current attribute in the clincaldata instance even though it is outside of the first node. It works as expected if the logic is changed from:
[position()=${pos}]
to: 
[position()=number(${pos})]

It also works correctly for [position()=1] and [position()='1']

**Comment by MartijnR** [comment]
@powen-oc, the clinical data file is accidentally the same as the form definition

**Comment by pbowen-oc** [comment]
@MartijnR - Sorry about that. It should be correct now.

**Comment by MartijnR** [comment]
```bash
curl --user enketorules: -d "server_url=http://localhost:3000&form_id=cross-form.openclinica.current&ecid=1&instance=<data xmlns:oc=\"http://openclinica.org/xforms\"  xmlns:enk=\"http://enketo.org/xforms\" ><ae_rep_count>3</ae_rep_count><ae_rep enk:last-used-ordinal=\"3\"  enk:ordinal=\"1\" ><pos>1</pos><pos_comment/></ae_rep><ae_rep   enk:ordinal=\"2\" ><pos>2</pos><pos_comment/></ae_rep><ae_rep   enk:ordinal=\"3\" ><pos>3</pos><pos_comment/></ae_rep><meta><instanceID>uuid:1234</instanceID><instanceID_comment/></meta><ae_rep_count_comment/></data>&instance_id=a" http://localhost:8005/oc/api/v1/instance/edit
```

**Comment by MartijnR** [comment]
`[number(ref)]` works correctly too

**Comment by MartijnR** [comment]
I believe this issue is the same as https://github.com/enketo/enketo-core/issues/818

* [ ] check

**Comment by MartijnR** [comment]
See cherry picking advice in #503. 

For master this fix will be included in the next version of Enketo Express (3.0.2 or 3.1.0)

**Comment by pbowen-oc** [comment]
Looks good in testing.
